lib/TickLib.sol

```diff
@@ -128,9 +129,8 @@ library LeafLib {
     return uint(raw << (OFFER_BITS * ((index * 2) + 1)) >> (256 - OFFER_BITS));
   }
 
-  // TODO optimize with a hashmap, or:
-  // if > half: +=1; if += quarter: +=1, or:
-  // or nested if dichotomy
+  // Will check for the first position (0,1,2 or 3) that has a nonzero first-of-tick or a nonzero last-of-tick offer. Leafs where only one of those is nonzero are invalid anyway.
+  // Offers are ordered msb to lsb
```

Comment:
This is unintuitive, since it's the opposite for levels 😅

Reply:
agreed but I couldn't figure out how to easily do it the other way around